src/rgw/rgw_auth_keystone.h

```diff
@@ -87,17 +87,17 @@ class SecretCache {
   std::map<std::string, secret_entry> secrets;
   std::list<std::string> secrets_lru;
 
-  Mutex lock;
+  std::mutex lock;
 
   const size_t max;
 
-  const uint64_t s3_token_expiry_length;
+  const utime_t s3_token_expiry_length;
 
   SecretCache()
     : cct(g_ceph_context),
-      lock("rgw::auth::keystone::SecretCache"),
+      lock(),
       max(cct->_conf->rgw_keystone_token_cache_size),
-      s3_token_expiry_length(300) {
+      s3_token_expiry_length(300, 0) {
```

Comment:
missed this in the first path;  this should probably be configurable?  I don't think that's a problem for now

Reply:
@mattbenjamin we did try to make this configurable during testing but that only led to segfaults, the proper way to add new config options is a bit opaque. Is there a good guide or example for that?

Reply:
Any progress on this?